projects/hslayers-app/src/hslayers-app/hslayers-app.module.ts

```diff
@@ -24,9 +24,16 @@ export class AppModule implements DoBootstrap {
         setTimeout(waitAppElement, 50);
         return;
       }
-      document.querySelectorAll('hslayers-app').forEach((el) => {
-        appRef.bootstrap(HslayersAppComponent, el);
-      });
+      const apps: HTMLElement[] = Array.from(
+        document.querySelectorAll('hslayers-app')
+      );
+      for (const el of apps) {
+        if (!el.dataset.init) {
+          appRef.bootstrap(HslayersAppComponent, el);
+          el.dataset.init = 'true';
+          return;
```

Comment:
what is the purpose of `return` here?

Reply:
To exit from the loop i.e. to prevent (in case of having 2 apps) first to be bootstraped twice and preventing bootstrap of the second app by setting el.dataset.init = 'true';.

This part might need refactor in case we want better control of lets call it 'app placement'. Now its just goes from up to down.